client/consensus/babe/src/lib.rs

```diff
@@ -347,7 +347,11 @@ impl Config {
 	{
 		trace!(target: "babe", "Getting slot duration");
 
-		let best_block_id = BlockId::Hash(client.usage_info().chain.best_hash);
+		let mut best_block_id = BlockId::Hash(client.usage_info().chain.best_hash);
+		if client.usage_info().chain.finalized_state.is_none() {
+			debug!(target: "babe", "No finalized state is available. Reading config from genesis");
```

Comment:
As I've already mentioned in other comments here, i don't think it would be correct to tie presence of state to fast sync. Substrate as a framework supports header-only sync. Until recently it was used in the light client. Even though light client support is now removed, the header-only sync is still supported and might be useful for some chains. Fast sync is another mode that uses header-only sync as well.